pkg/service/model_build_target_group.go

```diff
@@ -538,6 +585,36 @@ func (t *defaultModelBuildTask) buildTargetGroupBindingNodeSelector(_ context.Co
 	}, nil
 }
 
+func (t *defaultModelBuildTask) buildTargetGroupBindingServiceRef(_ context.Context, targetType elbv2api.TargetType,
+	port corev1.ServicePort) (elbv2api.ServiceReference, error) {
+	var serviceRef elbv2api.ServiceReference
+	if targetType != elbv2api.TargetTypeALB {
+		serviceRef = elbv2api.ServiceReference{
+			Name: t.service.Name,
+			Port: intstr.FromInt(int(port.Port)),
+		}
+	}
+	return serviceRef, nil
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupBindingIngressRef(_ context.Context, targetType elbv2api.TargetType,
+	port corev1.ServicePort) (elbv2api.IngressReference, error) {
+	var ingressRef elbv2api.IngressReference
```

Comment:
Do we check that the `ingressClassName` of that Ingress refers to an Ingress backed by this controller?

Reply:
There is nothing doing that currently. The `IngressRef` is only validated in the `buildTargetLoadBalancerArn` function in `resource_manager.go`, when we attempt to register the ALB as a target. If a matching ingress isn't found, then that function returns an error and no target is registered. Do you think we should also be checking for an `IngressClass` resource (with `controller: ingress.k8s.aws/alb`)? Is there a scenario that this would protect against? (Still wrapping my head around this project and related k8s topics)